app.js

```diff
@@ -1,77 +1,164 @@
+  for (var i = 0; i < productNames.length; i++) {
+    var productInfo = productNames[i];
+    var newItem = new Product(productInfo.name, productInfo.path);
+    allProducts.push(newItem);
+  }
+  console.log('allProducts: ', allProducts);
+}
 
-var previousImgs = [leftIndex, centerIndex, rightIndex];
-
-function randNum(){
-  return Math.floor(Math.floor(Math.random() * img.length));
+function randomNum(){
+  return Math.floor(Math.random() * productNames.length);
 };
 
-/*function displayPics() {
+function displayFinalResults() {
+  console.log('displayFinalResults');
+  for (var i = 0; i < allProducts.length; i++) {
+    var productClicksEl = document.createElement('li');
+    console.dir(productClicksEl);
+    clicks.appendChild(productClicksEl);
+    productClicksEl.textContent = allProducts[i].name + ': clicks ' + allProducts[i].clicks + ', views ' + allProducts[i].views;
+  }
+}
+
+function generateImg() {
   var leftIndex = randomNum();
   var centerIndex = randomNum();
   var rightIndex = randomNum();
-}*/
 
-var allProducts = [];
+  console.log('Left Index: ' + leftIndex);
+  console.log('Center Index: ' + centerIndex);
+  console.log('Right Index: ' + rightIndex);
+//left image generator
+  while(previousImgs.includes(leftIndex)){
+    var leftIndex = randomNum ();
+    console.log('New left: ' + leftIndex);
+  }
 
-var leftIndex = randNum();
-var leftImg = allProducts[leftIndex];
-previousImgs.push(leftIndex);
+  while(centerIndex === leftIndex || previousImgs.includes(centerIndex)){
+    var centerIndex = randomNum();
+    console.log('New Center:' + centerIndex);
+  }
 
-var centerIndex = randNum();
-while (centerIndex === leftIndex) {
-  centerIndex = randNum();
-}
-var centerImg = allProducts[centerIndex];
-previousImgs.push(centerIndex);
+  while(rightIndex === leftIndex || rightIndex === centerIndex || previousImgs.includes(rightIndex)) {
+    var rightIndex = randomNum();
+    console.log('New Right: ' + rightIndex);
+  }
+
+  previousImgs = [leftIndex, centerIndex, rightIndex];
+
+  console.log(previousImgs);
+
+  var leftImgElement = document.createElement('img');
+  leftImgElement.setAttribute('src', allProducts[previousImgs[0]].path);
+  leftImgElement.setAttribute('alt',allProducts[previousImgs[0]].name);
+  imgContainer.appendChild(leftImgElement);
+
+  var centerImgElement = document.createElement('img');
+  centerImgElement.setAttribute('src', allProducts[previousImgs[1]].path);
+  centerImgElement.setAttribute('id', allProducts[previousImgs[1]].name);
+  imgContainer.appendChild(centerImgElement);
 
-var rightIndex = randNum();
-while(rightIndex === leftIndex || rightIndex === centerIndex){ var rightIndex = randNum();
+  var rightImgElement = document.createElement('img');
+  rightImgElement.setAttribute('src', allProducts[previousImgs[2]].path);
+  rightImgElement.setAttribute('id', allProducts[previousImgs[2]].name);
+  imgContainer.appendChild(rightImgElement);
 }
+//returns previousImgs:
+
+createProducts();
```

Comment:
If you group function calls like these at the bottom of your file, it'll be a lot easier to find them in the future.